python/ray/worker.pyi

```diff
@@ -16,58 +17,64 @@ T9 = TypeVar("T9")
 R = TypeVar("R")
 
 
-class RemoteFunction(Generic[T0, T1, T2, T3, T4, T5, T6, T7, T8, T9]):
-    def __init__(self, function: Callable[[T0, T1, T2, T3, T4, T5, T6, T7, T8, T9], Any]) -> None: pass
+class RemoteFunction(Generic[R, T0, T1, T2, T3, T4, T5, T6, T7, T8, T9]):
+    def __init__(self, function: Callable[[T0, T1, T2, T3, T4, T5, T6, T7, T8, T9], R]) -> None: pass
 
     @overload
-    def remote(self) -> ObjectRef: ...
+    def remote(self) -> ObjectRef[R]: ...
     @overload
-    def remote(self, arg0: Union[T0, ObjectRef]) -> ObjectRef: ...
+    def remote(self, arg0: Union[T0, ObjectRef[R]]) -> ObjectRef[R]: ...
```

Comment:
These should all be independent return variables, or ObjectRefs of the same type. That is to say, `Union[T0, ObjectRef[T0]]`  -- because suppose this function is one from `str -> int` (atoi) -- and we can only take a `Union[str, ObjectRef[int]] -> ObjectRef[int]` -- which is clearly a mismatch

Reply:
nice catch